examples/web-server/README.md

```diff
@@ -1,34 +1,223 @@
+def provide_feedback(project_id: str, app_id: str, feedback: Feedback) -> EmailAssistantState:
+    return _run_through(project_id, app_id, dict(feedback=feedback.feedback))
+```
+
+This represents a simple but powerful architecture. We can continue calling these endpoints
+until we're at a "terminal" state, at which point we can always ask for the state.
+If we decide to add more input steps, we can modify the state machine and add more input steps.
+We are not required to hold state in the app (it is all delegated to Burr's persistence),
+so we can easily load up from any given point, allowing the user to wait for seconds,
+minutes, hours, or even days before continuing.
+
+As the frontend simply renders based on the current state and the next step, it will always
+be correct, and the user can always pick up where they left off. With Burr's telemetry capabilities
+they can debug any state-related issues, ensuring a smooth user experience.
+
+### Persistence
+
+Note that we never called out to databases. It all just magically worked.. This is all because we decouple the persistence
+layer from the web-call. The application will be persisted (to whatever database you want),
+by burr's plugin capabilities -- read more [here](https://burr.dagworks.io/concepts/state-persistence/).
+This greatly reduces the amount you have to think about when developing. As Burr persistence is
+pluggable, you can write to your own database with whichever schema you prefer, customizing
+the schema for your project or using a generic one (state is just a JSON object -- you can easily serialize/deseriealize it).
+
+### Additional concerns
+
+#### Scaling
+
+But [is this webscale](https://www.youtube.com/watch?v=b2F-DItXtZs)? As anything, it depends on how you implement it.
+Two factors determine the scalability of this system:
+
+1. database layer -- can the database support the volume of inputs/outputs?
+2. compute layer -- can the server run fast enough to keep up with the users?
+
+For the database layer, it depends largely on the underlying database, as well as the
+schema you use. That said, Burr makes it easier due to natural partitioning of the data
+into `application_id` and `partition_key` (the latter could be the `user ID`), allowing common
+operations (such as _give me all applications for X user_ and _give me the state of application Y_)
+simple if you index your state table on the application ID and `partition_key`.
+
+For the compute layer, you can simply scale horizontally. The only tricky aspect is ensuring state synchronization
+and locking. As we cached the application object, we could potentially get into a position
+in which the state is out of sync. To solve this, you can either:
+
+1. Use a locking method (either in the database) to ensure that only one server is running a given application at any point
+2. Use sticky sessions/sharding to ensure that a given user always hits the same server
+
+Or possibly both.
```

Comment:
```suggestion
1. Use a locking method (either in the database) to ensure that only one server is running a given application at any point
2. Use sticky sessions/sharding to ensure that a given user always hits the same server
3. Handle forking of state at the persistence layer with a custom implementation

Or possibly some combination of the above.
```

Reply:
Yeah thought about it but it feels messy/scary. Let's add it in and see what people use?